scrape/scrape.go

```diff
@@ -445,7 +445,7 @@ func (sp *scrapePool) Sync(tgs []*targetgroup.Group) {
 			switch {
 			case nonEmpty:
 				all = append(all, t)
-			case !t.discoveredLabels.IsEmpty():
```

Comment:
`discoveredLabels` was set in `NewTarget()`, with a value from `PopulateLabels()`.
`PopulateLabels()` returned empty `origLabels` in case of error, but then `NewTarget()` was not called.
Otherwise, it would return at least job, instance, etc.  

So I think the only way we can get here is through some logic error.  If that still exists, we should fix it, not figure out a new way to paper over it.